src/main.rs

```diff
@@ -1,98 +1,173 @@
-use std::convert::TryInto;
+use std::env;
 
-const BLOCK_SIZE: usize = 32; // Size of each block in bits
-const HASH_SIZE: usize = 32; // Size of the hash code in bits
+const BLOCK_SIZE: usize = 8; // Size of each block in bytes
+const HASH_SIZE: usize = 8; // Size of the hash code in bytes
+const INITIAL_STATE: [u8; HASH_SIZE] = [255, 100, 211, 37, 112, 167, 41, 37]; // randomly generated initial state
 
-struct XorHasher {
+struct EvanHash {
     state: [u8; HASH_SIZE],
-    block_count: usize,
+    block: [u8; BLOCK_SIZE],
+    length: usize,
+    data: Vec<u8>,
+    debug: bool,
```

Comment:
The logging function exists outside of this object. Extract `debug` from this struct where it does not belong.

Reply:
Probably should inject the logger instead of the setting.